src/check/property/SkipEqualValuesWithComparatorProperty.ts

```diff
@@ -0,0 +1,19 @@
+import { IRawProperty } from './IRawProperty';
+import { Random } from '../../random/generator/Random';
+import { Shrinkable } from '../arbitrary/definition/Shrinkable';
+
+export class SkipEqualValuesWithComparatorProperty<Ts, IsAsync extends boolean> implements IRawProperty<Ts, IsAsync> {
+  private coveredValues: Ts[] = [];
```

Comment:
Using a `Set` would be better in terms of performances. Array structure will be too costly for very high number of runs